crates/networking/p2p/rlpx/utils.rs

```diff
@@ -7,35 +7,46 @@ use k256::{
 };
 use sha3::{Digest, Keccak256};
 use snap::raw::{max_compress_len, Decoder as SnappyDecoder, Encoder as SnappyEncoder};
+use std::array::TryFromSliceError;
 use tracing::{debug, error, warn};
 
 pub fn sha256(data: &[u8]) -> [u8; 32] {
     use k256::sha2::Digest;
     k256::sha2::Sha256::digest(data).into()
 }
 
-pub fn sha256_hmac(key: &[u8], inputs: &[&[u8]], size_data: &[u8]) -> [u8; 32] {
+pub fn sha256_hmac(
+    key: &[u8],
+    inputs: &[&[u8]],
+    size_data: &[u8],
+) -> Result<[u8; 32], RLPEncodeError> {
     use hmac::Mac;
     use k256::sha2::Sha256;
 
-    let mut hasher = hmac::Hmac::<Sha256>::new_from_slice(key).unwrap();
+    let mut hasher = hmac::Hmac::<Sha256>::new_from_slice(key)
+        .map_err(|error| RLPEncodeError::Custom(error.to_string()))?;
     for input in inputs {
         hasher.update(input);
     }
     hasher.update(size_data);
-    hasher.finalize().into_bytes().into()
+    Ok(hasher.finalize().into_bytes().into())
 }
 
-pub fn ecdh_xchng(secret_key: &SecretKey, public_key: &PublicKey) -> [u8; 32] {
+pub fn ecdh_xchng(
+    secret_key: &SecretKey,
+    public_key: &PublicKey,
+) -> Result<[u8; 32], RLPDecodeError> {
     k256::ecdh::diffie_hellman(secret_key.to_nonzero_scalar(), public_key.as_affine())
         .raw_secret_bytes()[..32]
         .try_into()
-        .unwrap()
+        .map_err(|error: TryFromSliceError| RLPDecodeError::Custom(error.to_string()))
```

Comment:
In general, try to avoid using the Custom error, you can create a new error variant if needed.